app/plugin_install.go

```diff
@@ -103,10 +110,10 @@ func (a *App) installPlugin(pluginFile io.ReadSeeker, replace bool) (*model.Mani
 }
 
 func (a *App) RemovePlugin(id string) *model.AppError {
-	return a.removePlugin(id)
+	return a.removePlugin(id, false)
 }
 
-func (a *App) removePlugin(id string) *model.AppError {
+func (a *App) removePlugin(id string, synching bool) *model.AppError {
```

Comment:
Rather than a bool, maybe it would be cleaner and clearer to break this up into multiple functions. Remove from filestore, remove from disk, etc. Weather or not we are syncing doesn't seem like a concern of this function.

Reply:
Agreed. Will be refactoring into a different methods. New commit coming soon

Reply:
Refactored into:

- `removePlugin`: High level method that does all operations for removing plugin locally and any synch operations with the filestore (and will include cluster notifications when merged with the other [PR](https://github.com/mattermost/mattermost-server/pull/11611))
- `removePluginLocally`: Performs all local operations to remove the plugin